internal/pkg/cli/job_init.go

```diff
@@ -84,6 +86,44 @@ func newInitJobOpts(vars initJobVars) (*initJobOpts, error) {
 
 // Validate returns an error if the flag values passed by the user are invalid.
 func (o *initJobOpts) Validate() error {
+	if o.Name != "" {
+		if err := validateJobName(o.Name); err != nil {
+			return err
+		}
+	}
+	if o.DockerfilePath != "" {
+		if _, err := o.fs.Stat(o.DockerfilePath); err != nil {
+			return err
+		}
+	}
+	if o.Schedule != "" {
+		_, cronErr := cron.ParseStandard(o.Schedule)
+		rate, timeErr := time.ParseDuration(o.Schedule)
+		if cronErr != nil && timeErr == nil {
+			if rate.Seconds() != float64(int64(rate.Seconds())) {
+				return fmt.Errorf("schedule rate %s cannot be in units smaller than a second", o.Schedule)
+			} else if rate.Seconds() < 60 {
+				return fmt.Errorf("schedule rate %s must be greater than a minute", o.Schedule)
+			}
+		}
+		if cronErr != nil && timeErr != nil {
+			return fmt.Errorf("schedule value %s is invalid", o.Schedule)
+		}
+	}
+	if o.Timeout != "" {
+		timeout, err := time.ParseDuration(o.Timeout)
+		if err != nil {
+			return err
+		}
+		if timeout.Seconds() != float64(int64(timeout.Seconds())) {
+			return fmt.Errorf("timeout duration %s cannot be in units smaller than a second", o.Timeout)
+		}
+	}
+	if o.Retries != 0 {
+		if o.Retries < 0 {
```

Comment:
I think it suffices to check for negative numbers. 
```suggestion
	if o.Retries < 0 {
```